src/main/java/com/glodblock/github/inventory/FluidConvertingInventoryAdaptor.java

```diff
@@ -179,11 +185,47 @@ public ItemStack simulateSimilarRemove(int amount, ItemStack filter, FuzzyMode f
 
     @Override
     public boolean containsItems() {
-        if (invFluids != null && invFluids.getTankInfo(side) != null) {
-            for (FluidTankInfo tank : invFluids.getTankInfo(side)) {
-                FluidStack fluid = tank.fluid;
-                if (fluid != null && fluid.amount > 0) {
-                    return true;
+        if (invFluids != null && invFluids.getTankInfo(this.side) != null) {
+            List<FluidTankInfo[]> tankInfos = new LinkedList<>();
+            if (invFluids instanceof TileCableBus && ((TileCableBus) invFluids).getPart(this.side) instanceof PartP2PLiquids) {
+                // read other ends of p2p for blocking mode
+                PartP2PLiquids invFluidsP2P = (PartP2PLiquids) ((TileCableBus) invFluids).getPart(this.side);
+                try {
+                    Iterator<PartP2PLiquids> it = invFluidsP2P.getOutputs().iterator();
+                    boolean checkedInput = false;
+                    while (it.hasNext() || !checkedInput) {
+                        PartP2PLiquids p2p;
+                        if (it.hasNext()) {
+                            p2p = it.next();
+                        } else {
+                            p2p = invFluidsP2P.getInput();
+                            checkedInput = true;
+                        }
+
+                        if (p2p == invFluidsP2P || p2p == null)
+                            continue;
+
+                        try {
+                            Method getTarget = p2p.getClass().getDeclaredMethod("getTarget");
```

Comment:
please move all reflect stuff to `Ae2Reflect`

Reply:
done